pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,229 @@
+
+
+@app.errorhandler(PackageError)
+@app.errorhandler(ValidationError)
+def uncaught_exception_handler(exc):
+    return exception_response(str(exc), exc)
+
+
+@app.before_request
+def set_app_attrs_from_config():
+    current_app.install = Install(
+        current_app.config['DCOS_ROOT'],
+        current_app.config['DCOS_CONFIG_DIR'],
+        current_app.config['DCOS_ROOTED_SYSTEMD'],
+        manage_systemd=True,
+        block_systemd=False)
+    current_app.repository = Repository(
+        current_app.config['DCOS_REPO_DIR'])
+
+
+@app.before_request
+def create_work_dir():
+    os.makedirs(current_app.config['WORK_DIR'], exist_ok=True)
+
+
+@app.route('/repository/', methods=['GET'])
+def get_package_list():
+    return package_listing_response(current_app.repository.list())
+
+
+@app.route('/repository/<package_id>', methods=['GET'])
+def get_package(package_id):
+    return package_response(package_id, current_app.repository)
+
+
+@app.route('/repository/<package_id>', methods=['POST'])
+def fetch_package(package_id):
+    try:
+        repository_url = request.json['repository_url']
+    except Exception:
+        return (
+            error_response(
+                'Request body must be a json object with a `repository_url` '
+                'key.'
+            ),
+            http.client.BAD_REQUEST,
+        )
+
+    try:
+        actions.fetch_package(
+            current_app.repository,
+            repository_url,
+            package_id,
+            current_app.config['WORK_DIR'])
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.BAD_REQUEST,
+        )
+    else:
+        response = empty_response
+
+    return response
+
+
+@app.route('/repository/<package_id>', methods=['DELETE'])
+def remove_package(package_id):
+    try:
+        actions.remove_package(
+            current_app.install,
+            current_app.repository,
+            package_id)
+    except PackageNotFound:
+        response = (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageConflict:
+        response = (
+            error_response('Package {} is active, so it can\'t be removed.'),
+            http.client.CONFLICT,
+        )
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    else:
+        response = empty_response
+
+    return response
+
+
+@app.route('/active/', methods=['GET'])
+def get_active_package_list():
+    return package_listing_response(current_app.install.get_active())
+
+
+@app.route('/active/<package_id>', methods=['GET'])
+def get_active_package(package_id):
+    response = package_response(package_id, current_app.repository)
+
+    # Return early if there was an error loading the package.
```

Comment:
The comments at like 175 and 179 seem redundant to me. You could consider killing it.